### PR TITLE
Work on book model

### PR DESCRIPTION
**Pull request by kriazantseva**

### CODE REVIEW

File: src/main/java/mate/academy/bookapp/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookapp/controller/BookController.java]: use field injection

File: src/main/java/mate/academy/bookapp/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @GetMapping
+    public List getAll() {
```
Comment (Vasul-Pankiv) [inline src/main/java/mate/academy/bookapp/controller/BookController.java]: parametrize the list you are going to return from this method

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookapp/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/controller/BookController.java]:
```suggestion
@RequestMapping("/books")

```

File: src/main/java/mate/academy/bookapp/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+public class BookController {
+    @Autowired
+    private BookService bookService;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/controller/BookController.java]:
```suggestion
    private final BookService bookService;
```

and add `@RequiredArgsConstructor` under the controller class

File: src/main/java/mate/academy/bookapp/dto/BookDto.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
+public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/dto/BookDto.java]: Use java record for DTO

File: src/main/java/mate/academy/bookapp/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
+public class CreateBookRequestDto {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/dto/CreateBookRequestDto.java]: the same

File: src/main/java/mate/academy/bookapp/model/Book.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Table(name = "books")
+@Getter
+@Setter
+@ToString(exclude = {"description", "coverImage"})
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/model/Book.java]:
```suggestion
```

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -1 +1,11 @@
...
+spring.datasource.username=root
+spring.datasource.password=55FromTheInside
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.hibernate.ddl-auto=validate
+spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
+
+spring.jackson.deserialization.fail-on-unknown-properties=true
+
+liquibase.change-log=classpath:db/changelog/db.changelog-master.yaml
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties:12]:
```suggestion
liquibase.change-log=classpath:db/changelog/db.changelog-master.yaml

server.servlet.contextPath=/api
```

File: src/main/java/mate/academy/bookapp/repository/impl/BookRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/repository/impl/BookRepositoryImpl.java]: As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean only defines one constructor to begin with. However, if several constructors are available, at least one must be annotated to teach the container which one to use.

File: src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+import mate.academy.bookapp.dto.BookDto;
+import mate.academy.bookapp.dto.CreateBookRequestDto;
+import mate.academy.bookapp.mapper.BookMapper;
+import mate.academy.bookapp.model.Book;
+import mate.academy.bookapp.repository.BookRepository;
+import mate.academy.bookapp.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java:14]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java]:
```suggestion
        return bookRepository.findById(id)
                .map(bookMapper::toDto)
                .orElseThrow(() -> new EntityNotFoundException("Can`t find the book with id" + id));

```

File: src/main/resources/application.properties [hunk 12]
```diff
@@ -1 +1,11 @@
-spring.application.name=book-app
+spring.datasource.url=jdbc:mysql://localhost:3306/book_app?useSSL=false&serverTimezone=UTC
+spring.datasource.username=root
+spring.datasource.password=55FromTheInside
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.hibernate.ddl-auto=validate
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties:6]:
```suggestion
spring.jpa.hibernate.ddl-auto=validate
spring.jpa.show-sql=true
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/mate/academy/bookapp/repository/impl/BookRepositoryImpl.java (hunk 9 above)
Comment (MAMentorFX) [reply src/main/java/mate/academy/bookapp/repository/impl/BookRepositoryImpl.java]: not fixed

File: src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        return bookRepository.findById(id).map(bookMapper::toDto).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find the book with id" + id)
+        );
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java]: Mapstruct supports mapping collections from the box

File: src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        return bookRepository.findById(id).map(bookMapper::toDto).orElseThrow(
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookapp/service/impl/BookServiceImpl.java]:
```suggestion
        return bookRepository.findById(id)
                                         .map(bookMapper::toDto)
                                         .orElseThrow(
```

File: src/main/resources/application.properties (hunk 12 above)
Comment (MAMentorFX) [reply src/main/resources/application.properties:6]: not fixed

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**